syntax = "proto3";

import "common.proto";

service RingAllReduceService {
    rpc VariableWeightsInit(ResDictionary) returns (ResDictionary) {}
    rpc Recieve(RingAllReduceReq) returns (RingAllReduceResp) {}
}

message RingAllReduceReq {
    enum Stage {
        INIT = 0;
        SCATTER = 1;
        GATHER = 2;
    }
    Stage stage = 1;
    Gradients_Dictionary node_gradients = 2;
}


message Gradients_Dictionary {
  repeated ArrayRequest entry = 1;
  int32 acc_no=2;
}

message RingAllReduceResp {

}